### ISSUE TITLE
PHP warnings - Undefined array key

### ISSUE THREAD

**Issue by contaoacademy** [cliffparnitzky/FormDependentMandatoryField#27]
When I do a database migration, I get the following warnings:

```
PHP Warning:  Undefined array key "__selector__" in /public/cto4/vendor/cliffparnitzky/form-dependent-mandatory-field/system/modules/zFormDependentMandatoryField/forms/FormDependentMandatoryField.php on line 299
PHP Warning:  Undefined array key "default" in /public/cto4/vendor/cliffparnitzky/form-dependent-mandatory-field/system/modules/zFormDependentMandatoryField/forms/FormDependentMandatoryField.php on line 299
PHP Warning:  Undefined array key "textcustom" in /public/cto4/vendor/cliffparnitzky/form-dependent-mandatory-field/system/modules/zFormDependentMandatoryField/forms/FormDependentMandatoryField.php on line 299
PHP Warning:  Undefined array key "passwordcustom" in /public/cto4/vendor/cliffparnitzky/form-dependent-mandatory-field/system/modules/zFormDependentMandatoryField/forms/FormDependentMandatoryField.php on line 299
PHP Warning:  Undefined array key "textareacustom" in /public/cto4/vendor/cliffparnitzky/form-dependent-mandatory-field/system/modules/zFormDependentMandatoryField/forms/FormDependentMandatoryField.php on line 299
PHP Warning:  Undefined array key "hiddencustom" in /public/cto4/vendor/cliffparnitzky/form-dependent-mandatory-field/system/modules/zFormDependentMandatoryField/forms/FormDependentMandatoryField.php on line 299
```

**Comment by cliffparnitzky** [comment]
"passwordcustom" ... seems to be from another extension. Right?

**Comment by contaoacademy** [comment]
Hmm, I can't assign it to any extension at the moment. I'll check it again.

**Comment by contaoacademy** [comment]
Ahh, interesting. Your extension checks every field if there is a submit. The fields come from the core itself.

https://github.com/contao/contao/blob/050a082e300283afb6bdf741638dd7dff8939076/core-bundle/contao/dca/tl_form_field.php#L120

I close the ticket.

**Comment by contaoacademy** [comment]
Fritz helped me with the troubleshooting.
Here is the result.

---

The extension must take into account that it ignores `__selector__` and that some form fields also have a `*custom palette`.

This should be corrected here in the code:
https://github.com/cliffparnitzky/FormDependentMandatoryField/blob/96498d03d2ce769d3b1dfd310e605a5d8f43f414/system/modules/zFormDependentMandatoryField/dca/tl_form_field.php#L38-L44

**Comment by cliffparnitzky** [comment]
Thanks. How can I reproduce the probleme?

**Comment by contaoacademy** [comment]
I can give you my composer.json. So can use this setup.

Here I run `php vendor/bin/contao-console contao:migrate` an this will show the warnings.

**Comment by cliffparnitzky** [comment]
Fixed in cf8244d

**Comment by contaoacademy** [comment]
Thank you
